src/Http/Productions/ProductionSuggestionController.php

```diff
@@ -21,32 +23,54 @@ class ProductionSuggestionController
      */
     private $enrichedEventRepository;
 
+    /**
+     * @var Date|null
+     */
+    private $minDate;
+
     public function __construct(
         SimilaritiesClient $similaritiesClient,
-        DocumentRepositoryInterface $enrichedEventRepository
+        DocumentRepositoryInterface $enrichedEventRepository,
+        ?Date $minDate = null
     ) {
         $this->similaritiesClient = $similaritiesClient;
         $this->enrichedEventRepository = $enrichedEventRepository;
+        $this->minDate = $minDate;
     }
 
     public function nextSuggestion(): Response
     {
+        $date = $this->minDate ?: Date::now();
```

Comment:
Why don't we do this in the constructor? That way we can type the `minDate` variable to just `Date` instead of `?Date`.

Reply:
True, I'm a bit too careful when it comes to `Date::now()` but this is not part of a long-running process where this method would be called multiple times over a longer time period so that's a good improvement yes.

Reply:
Your suggestion is to do it in the `ControllerProvider` and make the `Date` in the constructor required, right?

Reply:
You can choose to either make the argument nullable and default to now() in the constructor or always pass a date, not much difference imho.

Reply:
Ah, yes I see. I was a bit confused because I misunderstood and thought you meant something like this with constructor:
![Screenshot 2020-09-03 at 13 31 54](https://user-images.githubusercontent.com/959026/92109870-0fb6f480-edea-11ea-9298-552292197320.png)

But you meant this right?
![Screenshot 2020-09-03 at 13 33 56](https://user-images.githubusercontent.com/959026/92109930-22312e00-edea-11ea-8017-5ed1526bb4b5.png)

Either way the result is the same as doing it in the ControllerProvider.